test: Add less restrictive gem spec for mongo gem

donoghuc opened the pull request:
It looks like latest features (including jruby 10 enablement) will require the latest minor release of the mongo gem.

Specifically:

The latest mongo gem relaxes the range for the ruby-bson gem. This would allow the uptake of https://github.com/mongodb/bson-ruby/pull/353 if it is accepted.

donoghuc committed:
Add less restrictive gem spec for mongo gem

It looks like latest features (including jruby 10 enablement) will require the
latest minor release of the mongo gem.

Specifically:

The latest mongo gem relaxes the range for the ruby-bson gem. This would allow
the uptake of https://github.com/mongodb/bson-ruby/pull/353 if it is accepted.

kaylareopelle commented on instrumentation/mongo/Appraisals:
```diff
@@ -5,5 +5,5 @@
 # SPDX-License-Identifier: Apache-2.0
 
 appraise 'mongo-2.13' do
```
Some of the mongo tests are failing on 3.4 with this update.

Seeing that the latest version is 2.21.0, what do you think about adding a new appraisal for the most recent version? 

Many instrumentation test suites have an appraisal without a version specified to make sure it's compatible with the latest release.

donoghuc replied:
Followed the pattern in other appraisal files (just doing major version when constrained to minor version). Pushed a commit to fix the tests, still digging in to impacts of those changes to generate a good changelog.

kaylareopelle replied:
Thanks!

donoghuc committed:
Update test expectations based on latest mongo gem

This commit updates the test expectation for an updated message string. The same
behavior is observbed, the message has just been updated.

This commit also updates construction of a test mongodb client with a newly
required value for auth_source when building a simple client.

arielvalentin committed:
Merge branch 'main' into jruby-10

arielvalentin committed:
Merge branch 'main' into jruby-10

kaylareopelle committed:
Merge branch 'main' into jruby-10
